src/uct/cuda/cuda_ipc/cuda_ipc_cache.c

```diff
@@ -125,6 +159,8 @@ UCS_PROFILE_FUNC(ucs_status_t, uct_cuda_ipc_cache_map_memhandle,
                       key->b_len, UCS_PGT_REGION_ARG(&region->super));
 
             *mapped_addr = region->mapped_addr;
+            ucs_assert(region->refcount < UINT_MAX);
+            region->refcount++;
```

Comment:
we are changing refcount under **read** lock, that seems racy
seems we need to make it atomic

Reply:
Thanks for the catch. Will fix this usage

Reply:
@yosefe Sorry for the delay in getting back on this. I've updated with your comments.